src/json-api-client/JsonaDataFormatter.ts

```diff
@@ -111,10 +115,56 @@ class JsonaDataFormatter {
         if (entity.attributes && Object.keys(entity.attributes).length === 0) {
             entity.attributes = undefined;
         }
+        let relationshipsWithoutAttributes;
+        if (entity.relationships) {
+            const relationships = entity.relationships;
+            relationshipsWithoutAttributes = this.removeAttributesFromRelationships(
+                relationships,
+            );
+            entity.relationships = relationshipsWithoutAttributes;
+        }
```

Comment:
This looks nice, but I think this might be better placed in its own function and called somewhere else. 
Reason behind this is because the method is called "removeClientGeneratedEntityFlag" and with this addition it definitely does a bit more than that.

Maybe this logic should be in separate function called something like `getEntityWithoutRelationshipAttributes`. 

Bear in mind that `removeClientGeneratedEntityFlag` and this new function should be called in tandem in certain places so a third function that wraps those to might be necessary to commit all those operations.